fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/FabricRegistryBuilder.java

```diff
@@ -50,31 +55,33 @@
 	 * @param <R> The registry type
 	 * @return An instance of FabricRegistryBuilder
 	 */
-	public static <T, R extends Registry<T>> FabricRegistryBuilder<T, R> from(R registry) {
+	public static <T, R extends MutableRegistry<T>> FabricRegistryBuilder<T, R> from(R registry) {
 		return new FabricRegistryBuilder<>(registry);
 	}
 
 	/**
 	 * Create a new {@link FabricRegistryBuilder} using a {@link SimpleRegistry}, the registry has the {@link RegistryAttribute#MODDED} attribute by default.
 	 *
-	 * @param type A class matching the type being stored in the registry
+	 * @param registryKey The RegistryKey to create the registry
+	 * @param lifecycle The {@link Lifecycle} used to create the registry
 	 * @param <T> The type stored in the Registry
 	 * @return An instance of FabricRegistryBuilder
 	 */
-	public static <T> FabricRegistryBuilder<T, SimpleRegistry<T>> createSimple(Class<T> type) {
-		return from(new SimpleRegistry<>());
+	public static <T> FabricRegistryBuilder<T, SimpleRegistry<T>> createSimple(class_5321<Registry<T>> registryKey, Lifecycle lifecycle) {
```

Comment:
Should there be helper methods here to allow for passing an `Identifier` rather than a `RegistryKey` here? And a default value for `lifecycle`?

Reply:
The RegistryKey now provides the type `T` removes the need for passing a Class<T> as well.

Reply:
That generic sugar in registry key is totally unused and can be safely casted. I support earthcomputer's proposal here. A needed registry key in this case can always be created by `RegistryKey.ofRegistry(id)`, creating a registry key of any type extending `Registry<T>` (T is a wildcard), with registry "minecraft:root" and value from id parameter.

Reply:
Also on lifecycle: we can use [`Lifecycle.stable()`](https://github.com/Mojang/DataFixerUpper/blob/41d097a3a2345588125bfef8231e962d51ec6ace/src/main/java/com/mojang/serialization/Lifecycle.java#L38) as the default lifecycle.

Reply:
I dont really think there is a need to pass the Lifecyle, stable should be fine for all.